lib/deno.js

```diff
@@ -208,5 +204,60 @@ export async function md5(str) {
   const MD5_URL = 'https://deno.land/std@0.106.0/hash/mod.ts';
   // @ts-ignore
   const { createHash } = await import(MD5_URL);
-  return createHash('md5').update(str).toString('base64');
+  return createHash('md5').update(str).toString('hex');
+}
+
+/**
+ * @param {Uint8Array} buffer
+ * @returns {Promise<string>}
+ */
+export async function base64Encode(buffer) {
+  const BASE64URL_URL = 'https://deno.land/std@0.106.0/encoding/base64.ts';
+  // @ts-ignore
+  const {encode} = await import(BASE64URL_URL)
+  return encode(buffer);
+}
+
+/**
+ * @param {string} str
+ * @returns {Promise<Uint8Array>}
+ */
+export async function base64Decode(str) {
+  const BASE64URL_URL = 'https://deno.land/std@0.106.0/encoding/base64.ts';
+  // @ts-ignore
+  const {decode} = await import(BASE64URL_URL)
+  return decode(str);
+}
+
+/**
+ * @param {string} hash
+ * @param {string | Uint8Array | ArrayBuffer} key
+ * @param {string | Uint8Array | ArrayBuffer} msg
+ * @param {string} [inputEncoding]
+ * @param {string} [outputEncoding]
+ * @returns {Promise<Uint8Array>}
+ */
+export async function hmacSign(hash, key, msg, inputEncoding, outputEncoding) {
+// Force lazy loading for users not using MD5
+  const HMAC_URL = 'https://deno.land/x/hmac@v2.0.1/mod.ts';
+  // @ts-ignore
+  const { hmac } = await import(HMAC_URL);
+  return hmac(hash, key, msg, inputEncoding, outputEncoding);
 }
+
+/**
+ * @param {"md2" | "md4" | "md5" | "ripemd160" | "ripemd160" | "sha1" | "sha224" | "sha256" | "sha384" | "sha512" | "sha3-224" | "sha3-256" | "sha3-384" | "sha3-512" | "keccak224" | "keccak256" | "keccak384" | "keccak512"} hash
+ * @param {Uint8Array} password
+ * @param {Uint8Array} salt
+ * @param {number} iterations
+ * @param {number} keyLen
+ * @returns {Promise<Uint8Array>}
+ */
+export async function pbkdf2Derive(hash, password, salt, iterations, keyLen) {
+  const PBKDF2_URL = 'https://deno.land/x/crypto@v0.10.0/pbkdf2.ts';
+  // @ts-ignore
+  const { pbkdf2 } = await import(PBKDF2_URL);
+  return pbkdf2();
+}
+
+export { Readable, Duplex } from "https://deno.land/std@0.106.0/node/stream.ts";
```

Comment:
0 dependencies by default would be great
```suggestion
async function getNodeStreams() {
  const STREAMS_URL = 'https://deno.land/std@0.106.0/node/stream.ts';
  const { Readable, Duplex } = await import(STREAMS_URL);
  return { Readable, Duplex };
}
```